### PR TITLE
packaging: Build coco-guest-components

### PR DESCRIPTION
**Pull request by fidencio**
Right now we're adding the support to build the coco-guest-components on our side, and also to provide the resulting tarball to the rootfs (although we're not doing that yet, and that will be done by #8870 later on).

In order to test it, please run a `make coco-guest-components-tarball` in your favourite architecture, as I could only test it out on x86_64.

There are more details in each commit message, so, please, take a look at those as well.

### CODE REVIEW

**Review by stevenhorsman — APPROVED** [review]
A couple of minor points, but the code looks good to me. Thanks

File: tools/packaging/scripts/lib.sh [hunk 1]
```diff
@@ -216,3 +216,11 @@ get_agent_image_name() {
 
 	echo "${BUILDER_REGISTRY}:agent-$(get_last_modification ${libs_dir})-$(get_last_modification ${agent_dir})-$(uname -m)"
 }
+
+get_coco_guest_components_image_name() {
+	libs_dir="${repo_root_dir}/src/libs"
+	agent_dir="${repo_root_dir}/src/agent"
+
+	coco_guest_components_script_dir="${repo_root_dir}/tools/packaging/static-build/coco-guest-components"
+	echo "${BUILDER_REGISTRY}:agent-$(get_from_kata_deps "externals.coco-guest-components.toolchain")-$(get_last_modification ${coco_guest_components_script_dir})-$(uname -m)"
```
[the remark this answers is not in the document]
Comment (fidencio) [reply tools/packaging/scripts/lib.sh:225]: Nice catch, thanks!  And now you know from where I got it. :-)

File: tools/packaging/scripts/lib.sh [hunk 2]
```diff
@@ -216,3 +216,11 @@ get_agent_image_name() {
 
 	echo "${BUILDER_REGISTRY}:agent-$(get_last_modification ${libs_dir})-$(get_last_modification ${agent_dir})-$(uname -m)"
 }
+
+get_coco_guest_components_image_name() {
+	libs_dir="${repo_root_dir}/src/libs"
+	agent_dir="${repo_root_dir}/src/agent"
```
[the remark this answers is not in the document]
Comment (fidencio) [reply tools/packaging/scripts/lib.sh:222]: Same here.

**Review by gkurz — APPROVED** [review]
LGTM. Thanks @fidencio !

**Pull request merged**